.github/workflows/build.yml

```diff
@@ -0,0 +1,102 @@
+name: moonlight-android
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: android-actions/setup-android@v3
+      with:
+        cmdline-tools-version: 9477386
+
+    - uses: actions/setup-java@v4
+      with:
+        java-version: 21
+        distribution: 'adopt'
+
+    - name: Set up environment
+      run: |
+        echo "ANDROID_NDK_HOME=$ANDROID_NDK_HOME" >> $GITHUB_ENV
+        echo "ANDROID_SDK_ROOT=$ANDROID_HOME" >> $GITHUB_ENV
+        echo "COMMIT_PREFIX=$(echo ${{ github.sha }} | cut -c1-8)" >> $GITHUB_ENV
+
+    - name: Checkout moonlight-android repo
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Decode debug keystore
+      env:
+        DEBUG_KEYSTORE: ${{ secrets.DEBUG_KEYSTORE }}
+      run: |
+        if [ -z "$DEBUG_KEYSTORE" ]
+        then
+          echo "No debug keystore value"
+        else
+          echo $DEBUG_KEYSTORE > debug.keystore.base64
+          base64 --decode debug.keystore.base64 > debug.keystore
+          sudo cp -f debug.keystore /root/.android/.
+        fi
+
+    - name: Build artifact
+      run: ./gradlew assemble
+
+    - name: Copy artifact
+      run: |
+        mkdir moonlight-android
+        cp app/build/outputs/apk/root/debug/app-root-debug.apk "moonlight-android/app-root-debug-${COMMIT_PREFIX}.apk"
+        cp app/build/outputs/apk/nonRoot/debug/app-nonRoot-debug.apk "moonlight-android/app-nonRoot-debug-${COMMIT_PREFIX}.apk"
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: moonlight-android
+        path: ${{ github.workspace }}/moonlight-android
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: github.ref == 'refs/heads/master'
+
+    steps:
+    - name: Checkout moonlight-android repo
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Download Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: dist
+        merge-multiple: true
+
+    - name: Re-zip artifacts
+      run: |
+        cd dist
+        COMMIT_PREFIX=$(echo ${{ github.sha }} | cut -c1-8)
+        for artifact in *.apk
+        do
+          file_name="${artifact%.apk}"
+          echo "-> Creating ${file_name//-${COMMIT_PREFIX}/}.zip"
+          zip "${file_name//-${COMMIT_PREFIX}/}.zip" "${file_name}.apk"
+        done
+
+    - name: Update Git Tag
+      run: |
+        git tag -f Pre-release
+        git push -f origin Pre-release
+
+    - name: Create Release
+      uses: ncipollo/release-action@v1
+      with:
+        prerelease: true
+        allowUpdates: true
```

Comment:
One issue with this approach is that once the release is first made, the tag will never be updated to point to a new commit. I dealt with this in Sunshine for a while when we had a "nightly" release. Not sure the best way to handle that.

Is the step above this supposed to take care of that?

Reply:
It is correct that the tag will be created once and have that timestamp, but the commit itself will be updated and show the latest commit message. What this does it will overwrite the previous one.